app/design/adminhtml/default/default/template/newsletter/template/edit.phtml

```diff
@@ -47,130 +47,188 @@
-        },
-
-        unStripTags: function () {
-            $('convert_button').show();
-            $('convert_button_back').hide();
-            $(this.id).value =  this.unconvertedText;
-            if(this.isEditor()) {
-                this.getEditor().turnOn();
-                this.getEditor().getToggleButton().show();
-            }
-            this.typeChange = false;
-            $('field_template_styles').show();
-            return false;
-        },
+const templateControl = {
+    unconvertedText: '',
+    typeChange: false,
+    templateName: false,
+    id: 'text',
 
-        save: function() {
-            if (this.typeChange) {
-                $('change_flag_element').value = '1';
-            }
-            if(this.isEditor()) {
-                tinymce.triggerSave();
-            }
-            templateForm.submit();
+    init() {
+        const convertButtonBack = document.getElementById('convert_button_back');
+        if (convertButtonBack) {
+            convertButtonBack.style.display = 'none';
+        }
+    },
+
+    stripTags() {
+        if (!window.confirm("<?= $this->jsQuoteEscape(Mage::helper('newsletter')->__('Are you sure that you want to strip all tags?')) ?>")) {
             return false;
-        },
+        }
 
-        saveAs: function() {
-            if (this.typeChange) {
-                $('change_flag_element').value = '1';
+        if (this.isEditor()) {
+            const editor = this.getEditor();
+            if (editor) {
+                editor.turnOff();
+                const toggleButton = editor.getToggleButton();
+                if (toggleButton) {
+                    toggleButton.style.display = 'none';
+                }
             }
+        }
 
-            if($F('code').blank() || $F('code')==templateControl.templateName) {
-               value = prompt('<?= $this->jsQuoteEscape(Mage::helper('newsletter')->__('Please enter new template name')) ?>', templateControl.templateName + '<?= $this->jsQuoteEscape(Mage::helper('newsletter')->__(' Copy')) ?>');
-               if(!value) {
-                   if(value !== null) {
-                       $('code').value = '';
-                       templateForm.submit();
-                   }
-                   return false;
-               } else {
-                   $('code').value = value;
-               }
-            }
+        const textElement = document.getElementById(this.id);
+        this.unconvertedText = textElement.value;
 
-            $('save_as_flag').value = '1';
+        document.getElementById('convert_button').style.display = 'none';
+        document.getElementById('convert_button_back').style.display = '';
 
-            if(this.isEditor()) {
-                tinymce.triggerSave();
+        // Strip HTML tags and scripts
+        const tempDiv = document.createElement('div');
+        tempDiv.innerHTML = textElement.value;
+        textElement.value = tempDiv.textContent || tempDiv.innerText || '';
```

Comment:
Since we have a helper function in `varien/js.js`:

```suggestion
        // Strip HTML tags and scripts
        textElement.value = stripTags(textElement.value);
```